idaes/core/property_base.py

```diff
@@ -198,10 +198,12 @@ def build_state_block(self, *args, **kwargs):
             for i in initialize.keys():
                 initialize[i]["parameters"] = self
 
-        return self.state_block_class(*args,
-                                      **kwargs,
-                                      default=default,
-                                      initialize=initialize)
+        return self.state_block_class(  # pylint: disable=not-callable
```

Comment:
For this case, I don't see any reason why we could not make `state_block_class` a method rather than a `property` - that way pylint should be able to tell that it is callable.

Reply:
One consequence that this change would have is that clients of `state_block_class` would have to use it as a callable (`state_block_class()`), i.e. breaking the current API. This would not only require changing all call sites (about 20 from a quick search), but also having to use the (IMO slightly awkward) double-parens syntax to call the class itself (e.g. `state_block = my_block.state_block_class()(**state_block_params)`).

Taken these into account, and considering that I don't think there's anything wrong with the current solution of using a callable through a property (a reasonable strategy to effectively "build" an instance-specific method at runtime), I can see the argument for leaving the pylint directive and avoid changing the code.